backend/database.js

```diff
@@ -84,6 +84,47 @@ export default (sbp('sbp/selectors/register', {
       }
     })
   },
+  'backend/db/streamEntriesBetween': async function (startHash: string, endHash: string, offset: number): Promise<*> {
+    offset = isNaN(parseInt(offset)) ? 0 : Math.max(0, parseInt(offset))
+
+    let prefix = '['
+    let isMet = false
+    let currentHEAD = endHash
+    let entry = await sbp('chelonia/db/getEntry', currentHEAD)
+    if (!entry) {
+      throw Boom.notFound(`entry ${currentHEAD} doesn't exist!`)
+    }
+    // NOTE: if this ever stops working you can also try Readable.from():
+    // https://nodejs.org/api/stream.html#stream_stream_readable_from_iterable_options
+    return new Readable({
+      async read (): any {
+        try {
+          entry = await sbp('chelonia/db/getEntry', currentHEAD)
```

Comment:
The way this function is written, it will still throw an exception because `currentHEAD` can be `null` at this point. And indeed that's what I'm still getting in the logs:

```
read(): Error during getEntry: no entry for null!: ChelErrorDBConnection: Error during getEntry: no entry for null!
```

Reply:
```
const previousHEAD = entry.message().previousHEAD
if (!currentHEAD || !previousHEAD || (isMet && !offset)) {
  this.push(']')
  this.push(null)
} else {
  currentHEAD = previousHEAD
}
```
Nice catch. I think this will fix it.